dashboard/app.py

```diff
@@ -0,0 +1,670 @@
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with release dates that the user selected
+    """
+    release_dates = st.sidebar.multiselect(
+        "Release Date:", options=df["release_date"].dt.date.unique())
+    return release_dates
+
+
+def build_sidebar_review_date(df: DataFrame) -> list:
+    """
+    Build sidebar with dropdown menu options to select review dates
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with review dates that the user selected
+    """
+    dates = st.sidebar.multiselect(
+        "Review Date:", options=df["review_date"].dt.date.unique())
+    return dates
+
+
+def build_sidebar_genre(df: DataFrame) -> list:
+    """
+    Build sidebar with dropdown menu options to select game genre
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A list with game genres that the user selected
+    """
+    dates = st.sidebar.multiselect(
+        "Genre:", options=df["genre"].unique())
+    return dates
+
+
+def build_sidebar_platforms() -> list:
+    """
+    Build sidebar with platform selection options
+
+    Returns:
+        list: A list with selected platforms
+    """
+    platforms = ["mac", "windows", "linux"]
+    selected_platforms = st.sidebar.multiselect(
+        "Compatible Platforms:", options=platforms, default=platforms)
+    return selected_platforms
+
+
+def build_sidebar_sentiment(df: DataFrame) -> tuple:
+    """
+    Build sidebar with slider option to select range for review sentiment
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+    Returns:
+        list: A tuple with minimum and maximum sentiment that the user has selected
+    """
+    sentiment = st.sidebar.slider(
+        "Sentiment:", min_value=0.0, max_value=5.0, value=(0.0, 5.0), step=0.1)
+    return sentiment
+
+
+def plot_games_release_frequency(df: DataFrame, titles: list[str], release_dates: list[datetime], review_dates: list[datetime],
+                                 genres: list[str], platforms: list[str], minimum_sentiment: float, maximum_sentiment: float) -> Chart:
+    """
+    Create a line chart for the number of games released per day
+
+    Args:
+        df (DataFrame): A pandas DataFrame containing all relevant game data
+
+        titles (list[str]):  A list with game titles that the user selected
+
+        release_dates (list[datetime]): A list with release dates that the user selected
+
+        review_dates (list[datetime]): A list with review dates that the user selected
+
+        genres (list[str]): A list with game genres that the user selected
+
+        platforms (list[str]): A list with selected platforms
+
+    Returns:
+        Chart: A chart displaying plotted data
+    """
+    if len(titles) != 0:
```

Comment:
what does it return if these things are equal to 0?

Reply:
Actually, also, if it is len = 0 you could just say `if not titles` instead of `if len(titles) != 0`

Just a slightly better practise but not a needed change

Reply:
It's to do with the filtering using the sidebar, if there is nothing in the list, it means no option has been selected so no filtering needs to be done. 

Angela's suggestion is correct also it ends up being `if titles`. Looks much better now.